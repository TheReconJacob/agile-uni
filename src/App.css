.searchContainer {
  display: flex;
  flex: 10;
  flex-direction: row;
}

.search {
  flex: 5;
}
