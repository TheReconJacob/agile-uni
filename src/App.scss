@import "sky-toolkit-core/all";
@import "sky-toolkit-ui/all";
@import "~@sky-uk/molecules/lib/SimpleMasthead";
@import "~sky-toolkit-ui/components/links";
@import "~sky-toolkit-ui/components/forms";
@import "~sky-toolkit-ui/components/buttons";
@import "~sky-toolkit-ui/components/typography"; // if using the `message` prop
@import "@sky-uk/toolkit-react/lib/ControlMessage"; // if using the `message` prop
@import "~sky-toolkit-ui/components/dropdown";
@import '~sky-toolkit-ui/components/hero';
@import '~sky-toolkit-ui/components/shine';
@import '~sky-toolkit-ui/components/typography';
@import '~sky-toolkit-ui/components/accordion';
@import '@sky-uk/toolkit-react/lib/Accordion';

.wrapper{
    min-height: calc(93vh - 72px);
    // min-height:100vh;
}