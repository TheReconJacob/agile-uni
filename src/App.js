import React from "react";
import "./App.scss";
import { AzureAD } from "react-aad-msal";
import { authProvider } from "./authProvider";
import Courses from "./pages/Courses";
import Home from "./pages/Home";
import Navbar from "./components/Navbar.js"
import Footer from "./components/Footer.js"
import { Route, BrowserRouter as Router } from "react-router-dom";
import Admin from "./pages/Admin";

const jwt = require('jsonwebtoken');

class App extends React.Component {
  constructor(props) {
    super(props);
    // We can add more state
    this.state = {
      admin: false,
    };

    this.getRoles().then(roles => {
      if (roles != null && roles.includes('admin')){
        this.setState({
          admin: true
        });
      }
    });
  }

  async getRoles(){
    const token = await authProvider.getIdToken();
    const idToken = token.idToken.rawIdToken;
    const decodedToken = jwt.decode(idToken);

    if (decodedToken.roles !== undefined) {
      console.log('Returning roles');
      return decodedToken.roles;
    } else {
      console.log('No specific role - general user')
      return null
    }
  }

  render() {
    let adminComponent;
    let adminAddCourse;

    if (this.state.admin) {
      adminComponent = <h1>HELLO ADMIN</h1>;
      adminAddCourse = <Route path="/admin" component={Admin} />;
    }
    return (
      <>
        <AzureAD provider={authProvider} forceLogin={true}>
          <Router>
            <Navbar />
            <div>
              <Route exact path="/" component={Home} />
              <Route path="/courses" render={() => <Courses adminStatus={this.state.admin} />} />
              {adminAddCourse}
            </div>
            {adminComponent}
            <Footer />
          </Router>
        </AzureAD>
      </>

    );
  }
}

export default App;
