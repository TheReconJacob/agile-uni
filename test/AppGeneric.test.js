import React from 'react';
import App from '../src/App.js';
import localStorageMock from '../src/localStorageMock'
import renderer from 'react-test-renderer';

beforeEach(() => {
    localStorageMock.setItem('msal.idtoken', 'eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6IkJCOENlRlZxeWFHckdOdWVoSklpTDRkZmp6dyJ9.eyJhdWQiOiJjMGZiNzliYS1iNzJjLTQ3YzEtOTEyYy00OGVlNmNiYWM5NzIiLCJpc3MiOiJodHRwczovL2xvZ2luLm1pY3Jvc29mdG9ubGluZS5jb20vNjhiODY1ZDUtY2YxOC00YjJiLTgyYTQtYTRlZGRiOWM1MjM3L3YyLjAiLCJpYXQiOjE1NzMyMTc2NzYsIm5iZiI6MTU3MzIxNzY3NiwiZXhwIjoxNTczMjIxNTc2LCJhaW8iOiJBVFFBeS84TkFBQUFmM2JoT2VMZDllNW1QNXg1WndDQzhBeXB2T1BiZFpndWpoeFEyUHRDR21NaklhM2ZjQnAwcHluRy9jdmE3MHZxIiwibmFtZSI6IkhhbGwsIEVtaWx5IChBc3NvY2lhdGUgU29mdHdhcmUgRGV2ZWxvcGVyKSIsIm5vbmNlIjoiNGUxYzU2YWYtMzdiOC00NmUzLWFkMGEtYzdhZjAzYmEzYTNiIiwib2lkIjoiZDU0ZTM2MjctN2RiNC00MDNhLTg4NzAtNGViZTliMDcyYzE1IiwicHJlZmVycmVkX3VzZXJuYW1lIjoiZW1pbHkuaGFsbEBza3kudWsiLCJzdWIiOiJLTlhuMVJZUEVadHZoU0dFRWwzYUo5eU52UnVEQ3JvRFpvaWZ2aXdlUEhnIiwidGlkIjoiNjhiODY1ZDUtY2YxOC00YjJiLTgyYTQtYTRlZGRiOWM1MjM3IiwidXRpIjoiMFg4XzFHX193MFduQW9BeDJja19BQSIsInZlciI6IjIuMCJ9.GmP-6FlC9uwA--uYNt0riK_D9RfUwn_ZCU0HUbbzxbmWlUhiAI98rjRzwv7DU5qeYDwVFUhnbZIawnNup1g4shn6Jv2pi5O8un-QqXa8lI20naL8fbN-pQhRTSnsjkZDqhKfDlrCYwnTYkiTQ3EaDRPL_OJZ3WjJE11OuRkUoXt2nw9xd8F2nlsg09XTh1ISOQNK1tuPRL_Cn5AnvcqOq1pkEvmlyr9A6PX_E7SBnZBcX-KWhHzK_Z2WGPQvJnyhTGQT6D9Al9Ml2OCh44cNLNIoDeO3bQFAiZh2VgmULKj7SO6OVaYHI6MZ9MNPXI1wJATll-5xoUgHHtdZvXbshw')

});

test('Admin component not visible to non admin', () => {
    const component = renderer.create(
        <App />
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});