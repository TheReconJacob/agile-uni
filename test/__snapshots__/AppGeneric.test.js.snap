// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin component not visible to non admin 1`] = `null`;
